mp4parse/src/lib.rs

```diff
@@ -1173,129 +1173,141 @@ fn read_flac_metadata<T: Read>(src: &mut BMFFBox<T>) -> Result<FLACMetadataBlock
     })
 }
 
-fn read_esds<T: Read>(src: &mut BMFFBox<T>) -> Result<ES_Descriptor> {
+fn find_descriptor<'a>(data: &'a [u8], esds: &mut ES_Descriptor) -> Result<()> {
```

Comment:
Yes, it doesn't need the lifetime here. I forgot to remove it during refactory.
Thanks for catching!